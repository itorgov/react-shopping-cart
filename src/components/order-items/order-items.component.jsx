import React from 'react';
import {useSelector} from "react-redux";

import {OrderItemsHeader} from "../order-items-header/order-items-header.component";
import {OrderItem} from "../order-item/order-item.component";

import styles from './order-items.module.scss';

export const OrderItems = () => {
    const items = useSelector(state => state.items);

    return (
        <div className={styles.items}>
            <OrderItemsHeader/>
            {
                items.length
                    ? items.map(item => <OrderItem key={item.id} item={item}/>)
                    : <span className={styles.empty}>Your cart is empty</span>
            }
        </div>
    );
};
