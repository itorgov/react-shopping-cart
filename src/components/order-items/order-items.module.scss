.items {
  display: flex;
  flex-direction: column;

  .empty {
    align-self: center;
    font-size: 2rem;
    margin-top: 4rem;
  }
}
