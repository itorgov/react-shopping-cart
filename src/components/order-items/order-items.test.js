import React from "react";
import {mount} from "enzyme";
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux';

import {OrderItems} from './order-items.component';

const mockStore = configureStore();

describe('<OrderItems />', () => {
    it('expect to render items', () => {
        const store = mockStore({
            cart: {
                items: [
                    {
                        id: 1,
                        name: "Test name",
                        price: 3900,
                        quantity: 2,
                    },
                ],
                note: '',
            }
        });

        expect(mount(<Provider store={store}><OrderItems /></Provider>)).toMatchSnapshot();
    });

    it('expect to render a message about the empty cart when there is no items', () => {
        const store = mockStore({
            cart: {
                items: [],
                note: '',
            }
        });

        expect(mount(<Provider store={store}><OrderItems /></Provider>)).toMatchSnapshot();
    });
});
