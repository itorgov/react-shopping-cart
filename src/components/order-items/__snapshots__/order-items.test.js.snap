// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItems /> expect to render a message about the empty cart when there is no items 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OrderItems>
    <div
      className="items"
    >
      <OrderItemsHeader>
        <div
          className="header"
        >
          <div>
            Product
          </div>
          <div>
            Price
          </div>
          <div>
            Qty
          </div>
          <div>
            Total
          </div>
          <div>
             
          </div>
        </div>
      </OrderItemsHeader>
      <span
        className="empty"
      >
        Your cart is empty
      </span>
    </div>
  </OrderItems>
</Provider>
`;

exports[`<OrderItems /> expect to render items 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OrderItems>
    <div
      className="items"
    >
      <OrderItemsHeader>
        <div
          className="header"
        >
          <div>
            Product
          </div>
          <div>
            Price
          </div>
          <div>
            Qty
          </div>
          <div>
            Total
          </div>
          <div>
             
          </div>
        </div>
      </OrderItemsHeader>
      <OrderItem
        item={
          Object {
            "id": 1,
            "name": "Test name",
            "price": 3900,
            "quantity": 2,
          }
        }
        key="1"
      >
        <div
          className="item"
        >
          <div>
            Test name
          </div>
          <div
            className="price"
          >
            <Price
              value={3900}
            >
              $39.00
            </Price>
          </div>
          <OrderItemQuantity
            item={
              Object {
                "id": 1,
                "name": "Test name",
                "price": 3900,
                "quantity": 2,
              }
            }
          >
            <div>
              <OrderItemButton
                onClick={[Function]}
                variant="decrement"
              >
                <button
                  className="button decrement"
                  onClick={[Function]}
                >
                  <ForwardRef(SvgMinus)>
                    <svg>
                      minus.svg
                    </svg>
                  </ForwardRef(SvgMinus)>
                </button>
              </OrderItemButton>
              <span
                className="quantity"
              >
                2
              </span>
              <OrderItemButton
                onClick={[Function]}
                variant="increment"
              >
                <button
                  className="button increment"
                  onClick={[Function]}
                >
                  <ForwardRef(SvgPlus)>
                    <svg>
                      plus.svg
                    </svg>
                  </ForwardRef(SvgPlus)>
                </button>
              </OrderItemButton>
            </div>
          </OrderItemQuantity>
          <div
            className="price"
          >
            <Price
              value={7800}
            >
              $78.00
            </Price>
          </div>
          <OrderItemButton
            onClick={[Function]}
            shake={true}
            variant="remove"
          >
            <button
              className="button remove shake"
              onClick={[Function]}
            >
              <ForwardRef(SvgTrash)>
                <svg>
                  trash.svg
                </svg>
              </ForwardRef(SvgTrash)>
            </button>
          </OrderItemButton>
        </div>
      </OrderItem>
    </div>
  </OrderItems>
</Provider>
`;
